lib/app/endpoint/home.rb

```diff
@@ -164,46 +171,45 @@ class Home < Endpoint
           end
         end
 
-        # Handles EHR Launch and any redirects
         get '/:id/:key/:endpoint/?' do
           instance = Inferno::Models::TestingInstance.get(params[:id])
-          halt 404 unless !instance.nil? && instance.client_endpoint_key == params[:key] && %w[launch redirect].include?(params[:endpoint])
+          halt 404 unless !instance.nil? && instance.client_endpoint_key == params[:key] && ['launch','redirect'].include?(params[:endpoint])
 
           sequence_result = instance.waiting_on_sequence
 
           if sequence_result.nil? || sequence_result.result != 'wait'
-            redirect "/#{base_path}/#{params[:id]}/?error=no_#{params[:endpoint]}"
+            redirect "/#{BASE_PATH}/#{params[:id]}/?error=no_#{params[:endpoint]}"
           else
-            klass = Inferno::Sequence::SequenceBase.subclasses.find do |x|
-              x.name.demodulize.start_with?(sequence_result.name)
-            end
+            klass = instance.module.sequences.find{|x| x.name.demodulize == sequence_result.name}
 
             client = FHIR::Client.new(instance.url)
-            client.use_dstu2
+            if instance.version == 'dstu2'
+              client.use_dstu2
+            end
             client.default_json
             sequence = klass.new(instance, client, settings.disable_tls_tests, sequence_result)
 
-            timer_count = 0
-            stayalive_timer_seconds = 20
+            timer_count = 0;
+            stayalive_timer_seconds = 20;
 
             stream do |out|
+
               EventMachine::PeriodicTimer.new(stayalive_timer_seconds) do
-                timer_count += 1
+                timer_count = timer_count + 1;
                 out << js_stayalive(timer_count * stayalive_timer_seconds)
               end
 
-              out << erb(:details, {}, instance: instance,
-                                       sequences_groups: Inferno::Sequence::SequenceBase.sequences_groups,
-                                       sequences: Inferno::Sequence::SequenceBase.ordered_sequences,
-                                       sequence_results: instance.latest_results,
-                                       tests_running: true)
+              out << erb(:details, {}, {instance: instance,
+                                        sequence_results: instance.latest_results,
+                                        tests_running: true}
+              )
 
               out << js_hide_wait_modal
               out << js_show_test_modal
               count = sequence_result.test_results.length
               sequence_result = sequence.resume(request, headers, request.params) do |result|
-                count += 1
-                out << js_update_result(sequence, result, count, sequence.test_count)
+                count = count + 1
```

Comment:
See above regarding self-assignment
```suggestion
                count += 1
```